lib/twitter/accounts.ex

```diff
@@ -0,0 +1,82 @@
+defmodule Twitter.Accounts do
+  @moduledoc """
+  The Accounts context.
+  """
+  alias Twitter.Repo
+  alias Twitter.Accounts.User
+
+  @doc """
+  create user: validate parameters and insert in the db
+  """
+  @spec create_user(map()) ::
+          {:ok, User.t()} | {:error, Ecto.Changeset.t(User.t())}
+  def create_user(attrs \\ %{}) do
+    %User{}
+    |> User.registration_changeset(attrs)
+    |> Repo.insert()
+  end
+
+  @doc """
+  get_user: returns user by id
+  """
+  @spec get_user(integer() | String.t()) :: User.t() | nil
```

Comment:
maybe better to use non_neg_integer here as opposed to integer. Try to be as specific in these types as possible.

Reply:
ok